### PR TITLE
Diarization

### PR DESCRIPTION
**Pull request by happyhuman**
The samples in this code demo diarization: Basically who said what (when there is more than just one person talking).

### CODE REVIEW

**Comment by happyhuman** [comment]
@nnegrey , @dizcology , @theacodes Can you please review this PR?

File: speech/cloud-client/transcribe_diarization.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+# limitations under the License.
+
+r"""Google Cloud Speech API sample that demonstrates how to request
+speaker diarization.
+
+Example usage:
+    python transcribe_diarization.py \
+        resources/Google_Gnome.wav
+    python transcribe_diarization.py \
+        gs://cloud-ml-api-e2e-testing/speech/stereo_audio.wav
```
Comment (nnegrey) [inline speech/cloud-client/transcribe_diarization.py]: `cloud-ml-api-e2e-testing` public bucket?

**Comment by happyhuman** [comment]
Good catch @nnegrey. I moved all the 4 beta samples to beta_snippets.py and simplified a few things.

**Comment by happyhuman** [comment]
@tswast Can you please review this PR?

File: speech/cloud-client/beta_snippets.py [hunk 2]
```diff
@@ -126,6 +127,36 @@ def transcribe_file_with_auto_punctuation(path):
...
+    print('Waiting for operation to complete...')
+    response = client.recognize(config, audio)
+
+    for i, result in enumerate(response.results):
+        alternative = result.alternatives[0]
+        print('-' * 20)
+        print('First alternative of result {}: {}'
+              .format(i, alternative.transcript))
+        print('Speaker Tag for the first word: {}'
+              .format(alternative.words[0].speaker_tag))
```
Comment (dizcology) [inline speech/cloud-client/beta_snippets.py]: does each `word` in `alternative.words` have other relevant information concerning diarization besides `speaker_tag`?  if so print them as well, if not please ignore this comment.

File: speech/cloud-client/beta_snippets_test.py [hunk 3]
```diff
@@ -42,3 +44,11 @@ def test_transcribe_file_with_auto_punctuation(capsys):
...
 
     assert 'Okay. Sure.' in out
+
+
+def test_transcribe_diarization(capsys):
+    transcribe_file_with_diarization(
+        os.path.join(RESOURCES, 'Google_Gnome.wav'))
+    out, err = capsys.readouterr()
+
+    assert 'OK Google stream stranger things from Netflix to my TV' in out
```
Comment (dizcology) [inline speech/cloud-client/beta_snippets_test.py]: are there two different speakers in this audio file?  if so assert something about the speaker_tags being returned correctly.

**Pull request merged**